File: packages/pensopay_android/android/src/main/kotlin/com/flutter/pensopay/networking/pensopayapi/pensopaylink/models/PPMandate.kt
package com.flutter.pensopay.networking.pensopayapi.pensopaylink.models

import org.json.JSONObject

class PPMandate: JSONObject() {

    var id: Int = 0
    var subscription_id: String = ""
    var mandate_id: String = ""
    var state: String = ""
    var facilitator: String = ""
    var callback_url: String? = null
    var link: String = ""
    var reference: String = ""
    var created_at: String = ""
    var updated_at: String = ""

}

File: packages/pensopay_ios/ios/Classes/Networking/Pensopay API/Pensopay Link/Models/PPMandate.swift
//
//  PPMandate.swift
//  PensopaySDK
//
//  Created on 07/28/2022
//  Copyright © 2022 Pensopay. All rights reserved.
//

import Foundation

public class PPMandate : Codable {

    // MARK: - Properties

    public var id: Int
    public var subscription_id: Int
    public var mandate_id: String
    public var state: String
    public var facilitator: String
    public var callback_url: String?
    public var link: String

    var _reference: String?

    public var reference: String {
        set { _reference = "\(newValue)" }
        get { return "\(_reference)" }
    }

    public var created_at: String
    public var updated_at: String
    
}